hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/HFileOutputFormat2.java

```diff
@@ -248,7 +248,6 @@ public void write(ImmutableBytesWritable row, V cell) throws IOException {
           tableNameBytes = Bytes.toBytes(writeTableNames);
         }
         String tableName = Bytes.toString(tableNameBytes);
-        Path tableRelPath = getTableRelativePath(tableNameBytes);
```

Comment:
That is a nice find.
BTW reading this part of code, I think so much of optimization we can do.  It might not be relevant as this change. Though good to do IMO
...
} else {
          tableNameBytes = Bytes.toBytes(writeTableNames);
        }
        String tableName = Bytes.toString(tableNameBytes);
...
Only in case of multiTable, we need create tableName  and tableNameBytes  again and again for every write. So we can make sure this is created at 1st and do not create every time for not multiTable case.

private Path getTableRelativePath(byte[] tableNameBytes) {
        String tableName = Bytes.toString(tableNameBytes);
        String[] tableNameParts = tableName.split(":");
        Path tableRelPath = new Path(tableName.split(":")[0]);
        if (tableNameParts.length > 1) {
          tableRelPath = new Path(tableRelPath, tableName.split(":")[1]);
        }
        return tableRelPath;
      }
split is done 3 times. We can just refer tableNameParts[0], tableNameParts[1] and other places.  
Can u pls include these kind of fixes also in PR?

Reply:
tableName split 3 times int  getTableRelativePath method has fixed in this PR 

i will do some other fixes  in next commit , the code there is a little mess